plugins/messagelayer/consensus_plugin.go

```diff
@@ -225,7 +226,9 @@ type OpinionGivers map[identity.ID]OpinionGiver
 // Query retrieves the opinions about the given conflicts and timestamps.
 func (o *OpinionGiver) Query(ctx context.Context, conflictIDs []string, timestampIDs []string) (opinions opinion.Opinions, err error) {
 	for i := 0; i < StatementParameters.WaitForStatement; i++ {
-		if o.view != nil {
+		// query statement only if it's been received within the last round interval
```

Comment:
query statement status only if any statement has been received within the last round interval. This is important. We do want to consider old statements IF the node has been active in the last round.

Reply:
I adjusted the comment to your suggestion.